### ISSUE TITLE
Popup warning if group policy does not have enough funds to execute the proposal

### ISSUE THREAD

**Issue by erikalogie** [regen-network/groups-ui#56]
In the case that a user does not have enough funds to execute a proposal, this popup should prevent them from proceeding: 

https://www.figma.com/file/Mqvg4sP0Gwk1Jn4svsmhDp/Groups-UI?node-id=4800%3A23395&t=Wm4hM29nlA3eP2ck-1

**Comment by blushi** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Regen-Registry-5f8998bec8958d000f4609e2/issues/regen-network/groups-ui/56?planning-poker) with Zenhub @haveanicedavid @ryanchristo

**Comment by ryanchristo** [comment]
Is this necessary? Keplr provides an insufficient fee if the account does not have enough to execute the proposal.

![Screenshot from 2023-06-05 14-31-23](https://github.com/regen-network/groups-ui/assets/12519942/5f2e213f-c0a5-4f9e-a46d-76510ccec037)

**Comment by ryanchristo** [comment]
We also have the toaster already implemented if another error occurs:

![Screenshot from 2023-06-05 14-32-50](https://github.com/regen-network/groups-ui/assets/12519942/3356d33b-8361-4c39-a2a2-c979abc9ed68)

**Comment by ryanchristo** [comment]
Seems a bit redundant and unnecessary to me. If the user does not have enough funds to execute the transaction, keplr does not allow them to execute the transaction and displays an insufficient funds warning.

**Comment by blushi** [comment]
I think the initial purpose of this was to warn the user in case he doesn't have enough funds for the msg(s) in a proposal to be executed successfully (ie sending tokens or staking).
But I do agree that the wording in the current design is quite misleading and you should still be able to execute the proposal if you want, even if the execution will fail...
cc/ @erikalogie

**Comment by erikalogie** [comment]
> I believe the design should be reworked and we probably do not need this in the MVP.

Yes I agree does not need to be MVP. I changed the button to say "execute anyways", maybe that helps.

**Comment by blushi** [comment]
I believe we might want to change the wording as well, because right now, it can be misunderstood and we might think it's only about fees like @ryanchristo did

**Comment by erikalogie** [comment]
> I believe we might want to change the wording as well, because right now, it can be misunderstood and we might think it's only about fees like @ryanchristo did

Maybe like this? "Sorry, there are not enough funds to execute this proposal’s spending or staking requirements"

**Comment by blushi** [comment]
Yeah that sounds better!
Should we remove this of the MVP epic and move it back to the product backlog then?
I believe the existing bugs in there might be more important to tackle first.

**Comment by ryanchristo** [comment]
> I think the initial purpose of this was to warn the user in case he doesn't have enough funds for the msg(s) in a proposal to be executed successfully (ie sending tokens or staking, which means it's only applicable for certain msgs).

Ah, that makes sense.

> I believe the design should be reworked and we probably do not need this in the MVP.

I agree. I think we can remove this from MVP.

> Maybe like this? "Sorry, there are not enough funds to execute this proposal’s spending or staking requirements"

This works but we should maybe try to word it in a way that it will apply to future actions as well. Maybe something along the lines of "Insufficient funds to execute the messages within the proposal."

**Comment by ryanchristo** [comment]
Also worth noting, in most cases this will be the group policy account, not the user account.

**Comment by erikalogie** [comment]
> This works but we should maybe try to word it in a way that it will apply to future actions as well. Maybe something along the lines of "Insufficient funds to execute the messages within the proposal."

Ok cool I changed the wording

**Comment by ryanchristo** [comment]
I would bump the estimate on this now as well having a better understanding. Given the need to calculate the amounts within the messages (could be multiple messages), I think this would be a 3 or 5.
